client/modules/IDE/components/Toolbar.jsx

```diff
@@ -36,6 +37,17 @@ class Toolbar extends React.Component {
     this.setState({ projectNameInputValue: event.target.value });
   }
 
+  handleProjectNameClick() {
+    if (this.canEditProjectName) {
+      this.props.showEditProjectName();
+      setTimeout(() => {
+        if (this.projectNameInput) {
```

Comment:
FYI, there is a newish feature in JavaScript which is a lifesaver when dealing with possibly undefined variables. It's called the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) - `?.`.  You can write `this.projectNameInput?.focus()` instead of an `if` statement.  When `this.projectNameInput` is `undefined` it will not call `focus()`.  I use `?.` all the time and I don't know how I lived without it!

Reply:
Thanks for the suggestion, I'll try updating it!